index.js

```
@@ -158,6 +159,36 @@ Object.defineProperty(File.prototype, 'relative', {
   }
 });
 
+Object.defineProperty(File.prototype, 'dirname', {
+  get: function() {
+    if (!this.path) throw new Error('No path specified! Can not get dirname.');
+    return path.dirname(this.path);
+  },
+  set: function(dirname) {
+    this.path = path.join(dirname, path.basename(this.path));
+  }
+});
+
+Object.defineProperty(File.prototype, 'basename', {
+  get: function() {
+    if (!this.path) throw new Error('No path specified! Can not get basename.');
+    return path.basename(this.path);
+  },
+  set: function(basename) {
+    this.path = path.join(path.dirname(this.path), basename);
+  }
+});
+
+Object.defineProperty(File.prototype, 'extname', {
+  get: function() {
+    if (!this.path) throw new Error('No path specified! Can not get extname.');
+    return path.extname(this.path);
+  },
+  set: function(extname) {
+    this.path = replaceExt(this.path, extname);
```

setters should also throw the errors on no path